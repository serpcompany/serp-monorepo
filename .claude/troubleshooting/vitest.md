https://vitest.dev/guide/common-errors.html