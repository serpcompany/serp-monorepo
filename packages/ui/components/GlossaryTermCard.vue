<template>
  <div>
    <nuxt-link
      :to="`/posts/${term.slug}/`"
      class="group hover:border-primary relative flex flex-col border-4 px-4 pt-4 pb-8"
    >
      <h2 class="text-xl font-semibold">{{ term.title }}</h2>
      <p class="mt-4 text-sm">{{ term.oneLiner }}</p>
    </nuxt-link>
  </div>
</template>

<script setup lang="ts">
import type { PostIndex } from '@serp/types/types';

defineProps<{
  term: PostIndex;
}>();
</script>
