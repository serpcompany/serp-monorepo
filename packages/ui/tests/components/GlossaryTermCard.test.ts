// tests/components/GlossaryTermCard.snapshot.test.ts
import { describe, expect, it } from 'vitest';
import GlossaryTermCard from '../../components/GlossaryTermCard.vue';
import ComponentRender from '../componentRender';

describe('GlossaryTermCard Snapshot', () => {
  it.each([
    [
      'with keyword provided',
      {
        props: {
          term: {
            slug: 'example-term',
            keyword: 'Example Keyword',
            title: 'Example Title',
            oneLiner: 'A brief overview of the term.'
          }
        }
      }
    ],
    [
      'without keyword (fallback to title)',
      {
        props: {
          term: {
            slug: 'example-term',
            keyword: '',
            title: 'Example Title',
            oneLiner: 'A brief overview of the term.'
          }
        }
      }
    ]
  ])(
    'renders %s correctly',
    async (desc: string, options: { props: unknown }) => {
      const html = await ComponentRender(
        `GlossaryTermCard ${desc}`,
        options,
        GlossaryTermCard
      );
      expect(html).toMatchSnapshot();
    }
  );
});
