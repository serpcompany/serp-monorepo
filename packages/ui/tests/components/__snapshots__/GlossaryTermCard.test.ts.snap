// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`glossaryTermCard Snapshot > renders with keyword provided correctly 1`] = `
<div>
  <a
    class="
      border-2
      flex
      flex-col
      group
      hover:border-primary
      pb-8
      pt-4
      px-4
      relative
    "
    href="/posts/example-term/"
  >
    <h2 class="
    font-medium
    text-sm
  ">Example Keyword</h2>
    <p class="
    mt-4
    text-sm
  ">A brief overview of the term.</p>
  </a>
</div>
`;

exports[`glossaryTermCard Snapshot > renders without keyword (fallback to title) correctly 1`] = `
<div>
  <a
    class="
      border-2
      flex
      flex-col
      group
      hover:border-primary
      pb-8
      pt-4
      px-4
      relative
    "
    href="/posts/example-term/"
  >
    <h2 class="
    font-medium
    text-sm
  ">Example Title</h2>
    <p class="
    mt-4
    text-sm
  ">A brief overview of the term.</p>
  </a>
</div>
`;
